lib/SensorFusion/src/ExtendedKalmanFilter.h

```diff
@@ -0,0 +1,119 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Implementation of the Extended Kalman Filter
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ *
+ *
+ * @{
+ */
+
+#ifndef EXTENDEDKALMANFILTER_H
+#define EXTENDEDKALMANFILTER_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include <ArduinoEigen.h>
+#include <stdint.h>
+#include "LinearKalmanFilter.h"
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+/** Number of states used in the state vector of the Extended Kalman Filter. */
+static const uint8_t NUMBER_OF_STATES = 3U;
```

Comment:
Everyone who includes this header, will get this static definition. Means it will exist several times.
You can use it inside the class, but keep it there public.